### PR TITLE
Fixes #1356: add aws composite collector scaffold

### LINKED ISSUE #1356: Introduce AWS composite collector scaffold with service plug-ins and partial-failure isolation
## Problem

The current AWS provider collector is IAM-role focused, which is correct for the existing V1 surface. The AWS machine identity platform needs a collector architecture that can add services incrementally without turning one collector into a large, brittle AWS crawler.

We need a composite AWS collector scaffold that keeps the existing IAM collector intact while making room for EC2, ECS, EKS, Lambda, Secrets Manager, KMS, Bedrock/AgentCore, runtime evidence, and future account/region fan-out.

## What to change

- Introduce an AWS composite collector/orchestrator that can run multiple service-specific collectors.
- Keep the existing IAM role collector as the first plug-in; do not rewrite its behavior unnecessarily.
- Define a service collector interface that can receive account/region/capability context and return raw assets plus source diagnostics.
- Support service-level partial failures so one failing AWS service does not automatically fail the entire scan when safe to continue.
- Preserve bounded retry/backoff/throttling behavior and make future service collectors able to classify retryable provider errors.
- Emit collector/source diagnostics with enough detail to show which service/account/region failed.
- Add deterministic ordering and stable raw asset IDs so scans remain diffable.
- Add tests with fake service collectors covering success, partial failure, duplicate raw assets, context cancellation, and deterministic output.
- Update docs to describe the composite collector and how future AWS service collectors should plug in.

## Acceptance criteria

- Existing IAM-only AWS scans keep working through the composite path.
- A fake second service collector can be added in tests without changing IAM collector internals.
- Service-level diagnostics are preserved and surfaced through the existing scan partial-failure path.
- Composite output is deterministic across collector order and repeated runs.
- Context cancellation stops child collectors cleanly.
- Tests prove one service collector can fail non-fatally while IAM output is still collected.

## Dependencies

Depends on #1354 and #1355.

Do not open a PR for this issue until both #1354 and #1355 are closed. The composite collector should use the expanded graph/model contracts instead of inventing temporary raw shapes that will be replaced immediately.

## Blocks

This issue should be closed before opening PRs for concrete AWS service collectors such as:

- EC2 instance profile collector
- ECS task role collector
- EKS Pod Identity/IRSA collector
- Lambda execution role collector
- Secrets/KMS resource collectors
- Bedrock/AgentCore collector
- runtime evidence collector

## Contributing requirements

Follow `CONTRIBUTING.md`:

- Branch from `dev`.
- Keep the PR focused on the collector scaffold, not full service expansion.
- Start the PR body with `Fixes #<this issue number>`.
- Include tests and validation output in the PR description.
- Update docs when collector architecture changes.
- Sign commits for DCO.
- Include the required AI disclosure fields in the PR description.

### PR DESCRIPTION
**Pull request by Oluwatobi-Mustapha**
Fixes #1356

## Summary
- Add a composable AWS collection layer and keep IAM behavior intact.
- Add service-level partial-failure handling in the composite collector (continue scan on non-fatal service errors).
- Add scope-aware diagnostics that include service and account/region context.
- Add deterministic, deduplicated asset aggregation across services.
- Update runtime and CLI builders to construct collectors through a shared AWS composite path.
- Add unit tests for success path, partial failure, duplicate/order stability, context cancellation and scope propagation.
- Update AWS collector documentation with design, service onboarding steps, and diagnostic contract.

## Testing
- `go test ./internal/providers/aws`
- `go test ./internal/runtime`
- `go test ./internal/cli`

(Note: local environment does not currently have Go installed, so these commands were not executed here.)

## AI-assisted
AI-assisted: No

---
## Summary by cubic
Adds a composable AWS collector scaffold that runs IAM first, dedupes and sorts assets, and wires into CLI/runtime via `NewAWSScanner` with account/region scope. Also fails the scan when all service collectors error to avoid silent empty results. Fixes #1356.

- **New Features**
  - Added `AWSCompositeCollector` to run service collectors in order; IAM preserved via `iamCollectorAdapter`.
  - Deterministic dedupe/order; immediate abort on cancellation/deadline.
  - Scope-aware diagnostics now include service/account/region and use `service_collection_failed` with `aws_<service>` naming.
  - CLI/runtime now use `NewAWSScanner` and pass `account_id`/`region`; tests assert composite wiring; docs updated.

- **Bug Fixes**
  - Composite returns an error when all service collectors fail (including IAM-only), preventing empty “success”.

<sup>Written for commit 977fd97cb4d612899ba09015ae55b560e53d1a49. Summary will update on new commits. <a href="https://cubic.dev/pr/identrail/identrail/pull/1371?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: internal/providers/aws/composite_collector.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+		for _, sourceError := range serviceSourceErrors {
+			sourceErrors = append(sourceErrors, enrichSourceError(scope, sourceError))
+		}
+
+		if err != nil {
+			if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+				return nil, nil, err
+			}
+			sourceErrors = append(sourceErrors, sourceServiceFailureDiagnostic(scope, err))
+		}
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/providers/aws/composite_collector.go:126]: Addressed in 6bc57d4a. The composite collector now tolerates partial failures only when at least one service collector succeeds. If every attempted collector fails (including the IAM-only path), it aggregates the errors via `errors.Join` and returns a non-nil error, so the scan fails instead of reporting an empty partial success. Added `TestAWSCompositeCollectorAllServicesFailingReturnsError` and `TestAWSCompositeCollectorIAMOnlyFailureReturnsError` to lock this in.
